### PR TITLE
NFT configuration logger

### PR DESCRIPTION
**Pull request by tobowers**
This logger contract lets the owner of an erc1155 set arbitrary key/value pairs for the NFT. It's only *safe* for non-fungible ERC1155 assets though as any owner of a tokenID can set the values.

### CODE REVIEW

File: packages/kasumah-logger/contracts/KasumahOwnerConfigs.sol [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import "@openzeppelin/contracts/token/ERC1155/IERC1155.sol";
+
+// import "hardhat/console.sol";
+
+/**
+    @dev KasumahOwnerConfigs is a contract that lets the current owner of an 1155
+    set arbitrary key/value pairs for the asset. Warning here that for *fungible*
+    1155 tokens, *any* owner will be able to set the k/v pairs.
+ */
+contract KasumahOwnerConfigs is Context {
```
Comment (alabobriggs) [inline packages/kasumah-logger/contracts/KasumahOwnerConfigs.sol:14]: contract is good

File: packages/kasumah-logger/package.json [hunk 2]
```diff
@@ -16,7 +16,7 @@
   },
   "scripts": {
     "build": "./scripts/release.sh",
-    "typechain": "typechain 'artifacts/!(build-info)/**/+([a-zA-Z0-9]).json' --target ethers-v5",
+    "typechain": "typechain '{artifacts/!(build-info)/**/+([a-zA-Z0-9]).json,./node_modules/@openzeppelin/contracts/build/contracts/ERC1155PresetMinterPauser.json}' --target ethers-v5",
```
Comment (alabobriggs) [inline packages/kasumah-logger/package.json:19]: Why is this required?

File: packages/kasumah-logger/test/KasumahOwnerConfigs.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+function advanceBlock() {
+  return ethers.provider.send("evm_mine", []);
+}
+
+function bufferFromRet(ret: string) {
+  return Buffer.from(ret.slice(2), "hex");
+}
+
+describe("KasumahOwnerConfigs", () => {
```
Comment (alabobriggs) [inline packages/kasumah-logger/test/KasumahOwnerConfigs.ts:20]: test is good to go

**Review by alabobriggs — APPROVED** [review]
Amazing feature

File: packages/kasumah-logger/package.json (hunk 2 above)
Comment (tobowers) [reply packages/kasumah-logger/package.json:19]: for the tests - just so we can deploy an erc1155 for them.
